### ISSUE TITLE
3.1.1 is breaking class:list in Astro?

### ISSUE THREAD

**Issue by michaelpumo** [ACP-CODE/astro-favicons#46]
I'm using Astro `4.16.17` and I upgraded `astro-favicons` to `3.1.1` but then, for some reason, any use of `class:list` in Astro, that contains an angle bracket, broke my page/HTML.

e.g.

```
<div class:list={['w-full', '[&>img]:rounded-12']}>
```

See the screenshot:

<img width="757" alt="Screenshot 2024-12-12 at 23 11 20" src="https://github.com/user-attachments/assets/fe06d719-4900-48ec-a605-eacda1e5b46a" />

It appeared to be treating the first occurance of the `>` character as the end of the HTML tag itself.

I'm not sure if this is your package directly, or something it depends on? If I downgrade to `2.1.8` of `astro-favicons`, everything works as expected.

Thanks

**Comment by JakiChen** [comment]
Heyyy, @michaelpumo 

Thanks for the feedback, do you try setting `withCapo` to false to see if the bug still persists and give me feedback?

```ts
// @ts-check
import { defineConfig } from "astro/config";
import favicons from "astro-favicons";

export default defineConfig({
  integrations: [
    favicons({
      name: "",
      short_name: "",
      withCapo: false
    })
  ],
});
```

The new version of HTML content is not fundamentally different, as it was previously done by automatically inserting HTML into the previous via Vite's transform </head> .

Manually insert HTML into the head and view the HTML content

e.g `~/components/Meta.astro`

```ts
---
import { localizedHTML } from 'astro-favicons/middleware';
const favicons = localizedHTML(Astro.currentLocale);
console.log(favicons);
---
<meta charset="utf-8" />
<Fragment set:html={favicons} />
```


After 3.1.0, you can also manually place HTML anywhere in the meta, and by default, the HTML is automatically inserted into the `<head>` via the astro middleware with `astro-capo` core code. If there's an issue, I might want to consider removing the withCapo feature or troubleshooting the issue.

**Looking forward to your feedback, thanks**

**Comment by JakiChen** [comment]
@michaelpumo 

The problem should be here

`[&>img]:rounded-12`

In order to support the manual import of favicons tags between the heads, part of the content is rewritten, and the html is parsed into AST and inserted into the document.

In order to enjoy the features after 3.0, it is recommended that you install 3.0.6 to experience the new features.

Happy Coding！

**Comment by michaelpumo** [comment]
Thank you, I will try again with capo set to false.

Just an FYI though...this could break a lot of people's stuff because many people are using Tailwind now with arbitrary rules like this: `[&>img]:rounded-12`

Maybe a note needs to be added to the docs?

Thanks!

**Comment by JakiChen** [comment]
This issue should be resolved in [3.1.2](https://github.com/ACP-CODE/astro-favicons/releases/tag/v3.1.2), see the changelog.

If there are other problems with the tailwind-related symbol renderSync in the future, you can simply disable `withCapo`.

Thanks again for your feedback,
